src/base/QXmppHttpUploadIq.cpp

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (C) 2008-2019 The QXmpp developers
+ *
+ * Authors:
+ *  Linus Jahn <lnj@kaidan.im>
+ *
+ * Source:
+ *  https://github.com/qxmpp-project/qxmpp
+ *
+ * This file is a part of QXmpp library.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ */
+
+#include <QDomElement>
+#include <QMimeDatabase>
+
+#include "QXmppHttpUploadIq.h"
+#include "QXmppConstants_p.h"
+
+class QXmppHttpUploadRequestIqPrivate
+{
+public:
+    QString fileName;
+    qint64 size;
+    QMimeType contentType;
+};
+
+QXmppHttpUploadRequestIq::QXmppHttpUploadRequestIq()
+    : d(new QXmppHttpUploadRequestIqPrivate())
+{
+}
+
+QXmppHttpUploadRequestIq::~QXmppHttpUploadRequestIq()
+{
+    delete d;
+}
+
+/// Returns the file name of the file to be uploaded.
+
+QString QXmppHttpUploadRequestIq::fileName() const
+{
+    return d->fileName;
+}
+
+/// Sets the file name. The upload service will use this to create the upload/
+/// download URLs. This may also differ from the actual file name to get a
+/// different URL. It's not required to replace special characters (this is the
+/// server's job).
+
+void QXmppHttpUploadRequestIq::setFileName(const QString &fileName)
+{
+    d->fileName = fileName;
+}
+
+/// Returns the file's size in bytes.
+
+qint64 QXmppHttpUploadRequestIq::size() const
```

Comment:
I'm sorry for pointing out these things one by one.. but why fileName() vs size() ? I would expect either name() and size() or fileName() and fileSize().

QXmppTransferFileInfo use name() and size()

Reply:
Yeah right, I took the XML attr. names.

Reply:
Is that ok or should I better use consistent names?